app/src/androidTest/java/ch/epfl/culturequest/backend/LocalStorageTest.java

```diff
@@ -0,0 +1,116 @@
+            fragmentTestRule.getFragment().localStorage.storeImageLocally(bitmap, false);
+        } catch (Exception e) {
+            fail("Should not have thrown any exception");
+        }
+        int finalCount = countPendingImagesInSharedStorage();
+        assertThat(finalCount, is(initialCount + 1));
+    }
+
+    @Test
+    public void storeImageLocallyWithWifiStoresOneReadyImageInSharedStorage() {
+        int initialCount = countReadyImagesInSharedStorage();
+        Bitmap bitmap = BitmapFactory.decodeResource(getApplicationContext().getResources(), R.drawable.joconde);
+        try {
+            fragmentTestRule.getFragment().localStorage.storeImageLocally(bitmap, true);
+        } catch (Exception e) {
+            fail("Should not have thrown any exception");
+        }
+        int finalCount = countReadyImagesInSharedStorage();
+        assertThat(finalCount, is(initialCount + 1));
+    }
+
+    @Test
+    public void storeImageLocallyWithNullBitmapThrowsIOException() {
+        IOException exception = assertThrows(IOException.class, () -> {
+            fragmentTestRule.getFragment().localStorage.storeImageLocally(null, false);
+        });
+        assertThat(exception.getMessage(), is("Failed to save image."));
+    }
+
+
+    @Test
+    @After
+    public void deleteAllImagesInSharedStorage() {
+        Uri collection = fragmentTestRule.getFragment().localStorage.contentUri;
+
+        ContentResolver contentResolver = getApplicationContext().getContentResolver();
+        contentResolver.delete(collection, null, null);
+        int totalCount = countPendingImagesInSharedStorage() + countReadyImagesInSharedStorage();
+        assertThat(totalCount, is(0));
+    }
+
+    private int countPendingImagesInSharedStorage() {
+        Uri collection = fragmentTestRule.getFragment().localStorage.contentUri;
+
+        String selection = MediaStore.Images.Media.DISPLAY_NAME + " LIKE ?";
+        String[] selectionArgs = new String[]{"pending_%"};
+
+        // Counts the number of ready images (not pending) in the shared storage
+        try (Cursor cursor = getApplicationContext().getContentResolver().query(
+                collection,
+                null,
+                selection,
+                selectionArgs,
+                null)) {
+            return cursor.getCount();
+        }
+    }
+
+    private int countReadyImagesInSharedStorage() {
+        Uri collection = fragmentTestRule.getFragment().localStorage.contentUri;
+
+        String selection = MediaStore.Images.Media.DISPLAY_NAME + " NOT LIKE ?";
+        String[] selectionArgs = new String[]{"pending_%"};
+
+        // Counts the number of ready images (not pending) in the shared storage
+        try (Cursor cursor = getApplicationContext().getContentResolver().query(
+                collection,
+                null,
+                selection,
+                selectionArgs,
+                null)) {
+            return cursor.getCount();
+        }
+    }
```

Comment:
It may be better to refactor these 2 functions and just add a parameter to choose between pending/ready

Reply:
Ok I will do that :)